gapic/src/main/com/google/gapid/models/Devices.java

```diff
@@ -57,11 +71,19 @@
   private List<Device.Instance> replayDevices;
   private Device.Instance selectedReplayDevice;
   private List<DeviceCaptureInfo> devices;
+  private DeviceValidationInfo deviceValidationInfo;
+  private List<DeviceCaptureInfo> validatedDevices;
```

Comment:
This is the local list of validated devices. The problem with using the list (created from the proto) is that if you call the contains function, it defaults to == check instead of calling equals() check. == checks if the two protobuf messages are the same object while equals() checks if two protobuf messages have the same contents. When we call contains(), Java List by default calls ==, while ArrayList calls equals(). It doesn't make sense to create a new arraylist every time we want to compare, which is why I created a member directly. 

Also, this helps in writing the cache back to the file. We load the cache into this list, add any new entries to the list if needed, write the list back to the file.

Reply:
Done.